tests/testament/categories.nim

```diff
@@ -63,6 +64,52 @@ proc compileRodFiles(r: var TResults, cat: Category, options: string) =
   test "gtkex1", true
   test "gtkex2"
 
+# --------------------- flags tests -------------------------------------------
+
+proc flagTests(r: var TResults, cat: Category, options: string) =
+  template test(filename: untyped, clearCacheFirst=false) =
+    if clearCacheFirst: delNimCache(filename, options)
+    testSpec r, makeTest(flagsDir / filename, options, cat)
+
+  # --genscript
+  test "tgenscript", true
+  test "tgenscript"
+
+  var build = ""
+  var run = ""
+  var cmd = "cd " & flagsDir / "nimcache" & " && "
```

Comment:
Don't use ``&&`` and string munging. Call ``execShellCmd`` once or twice instead.